### DISCUSSION TITLE
Dynamic array of values

### DISCUSSION THREAD

**Discussion by iakshay** [edmundhung/conform#573] [Q&A] [answered]
Hi, is it possible to add an item to an array field. It's not clear to me, how I can add an items to `todos` (from the docs example)

Here is how remix-validated-form does it - https://www.remix-validated-form.io/reference/use-field-array

**Comment by edmundhung** [comment, marked as answer]
You will need to use the [insert](https://conform.guide/intent-button#form-controls) intent to add an item. There are both a declarative version that works before JS or an imperative version:

```tsx
// Declarative
<button
  type="submit"
  {...form.insert.getButtonProps({ name: fields.todos.name, defaultValue: '' })
/>

// imperative
<button
  type="button"
  onClick={() => form.insert({ name: fields.todos.name, defaultValue: '' })}
/>
```

You can also find a working example here: https://stackblitz.com/github/edmundhung/conform/tree/main/examples/remix

**Reply by iakshay** [reply]
Thanks, I somehow missed this in examples. Would be nice to have a document example for dynamic updates.
